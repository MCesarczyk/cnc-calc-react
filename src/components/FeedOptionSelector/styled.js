import styled from "styled-components";

export const FormLabel = styled.label`
    display: flex;
    gap: 5px;
    align-items: center;
    justify-content: space-between;

    @media (max-width: 480px) {
        flex-direction: column;
        justify-content: center;
    }
`;

export const RadioButtonContainer = styled.div`
    display: flex;
    gap: 10px;
`;