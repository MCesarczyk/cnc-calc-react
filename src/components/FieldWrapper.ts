import styled from 'styled-components';

export const FieldWrapper = styled.div`
  display: flex;
  gap: 0.5rem;
  align-items: center;
  justify-content: space-between;

  @media (max-width: ${({ theme }) => theme.breakpoint.xs}) {
    flex-direction: column;
    justify-content: center;
    gap: 0.25rem;
  }
`;
