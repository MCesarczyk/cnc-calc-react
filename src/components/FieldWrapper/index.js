import styled from "styled-components";

export const FieldWrapper = styled.div`
    display: flex;
    gap: 0.5rem;
    align-items: center;
    justify-content: space-between;

    @media (max-width: 480px) {
        flex-direction: column;
        justify-content: center;
        gap: 0.25rem;
    }
`;