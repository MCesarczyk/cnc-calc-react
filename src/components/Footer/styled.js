import styled from "styled-components";

export const FooterWrapper = styled.footer`
  text-align: center;
  margin: 0;
`;

export const FooterContent = styled.p`
  color: ${({ theme }) => theme.color.primary};
  padding: 0 25px;
  font-size: 14px;
`;