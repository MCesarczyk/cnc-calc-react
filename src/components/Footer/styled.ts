import styled from "styled-components";

export const FooterWrapper = styled.footer`
  text-align: center;
  margin: 0;
`;

export const FooterContent = styled.p`
  color: ${({ theme }) => theme.color.primary};
  padding: 0 25px;
  font-size: 14px;
`;

export const FooterLink = styled.a`
  color: inherit;
  transition: 0.5s ease-out;

  &:hover {
    color: crimson;
  }
`