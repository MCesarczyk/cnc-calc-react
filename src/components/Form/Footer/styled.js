import styled from "styled-components";

export const ButtonsContainer = styled.div`
    max-width: 30rem;
    margin: 0 auto;
    display: flex;
    justify-content: center;
    gap: 0.75rem;

    @media (max-width: ${({ theme }) => theme.breakpoint.xs}) {
        flex-direction: column;
        align-items: stretch;
    }
`;