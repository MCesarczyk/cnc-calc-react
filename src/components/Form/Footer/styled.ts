import styled, { css } from 'styled-components';

export const ButtonsContainer = styled.div`
  max-width: 30rem;
  margin: 0 auto;
  display: flex;
  justify-content: center;
  gap: 0.75rem;

  @media (max-width: ${({ theme }) => theme.breakpoint.xs}) {
    flex-direction: column;
    align-items: stretch;
  }
`;

export const FooterButton = styled.button`
  display: flex;
  min-width: 12rem;
  width: 100%;
  justify-content: center;
  color: ${({ theme }) => theme.color.primaryContrast};
  background-color: ${({ theme }) => theme.color.primary};
  border: solid ${({ theme }) => theme.color.primary};
  padding: 0.25rem;
  border-radius: 1.5rem;

  &:hover {
    filter: brightness(150%);
  }

  &:active {
    filter: brightness(200%);
    border-color: currentColor;
  }

  ${({ type }) =>
    type === 'submit' &&
    css`
      color: ${({ theme }) => theme.color.submitButtonFont};
    `}

  ${({ type }) =>
    type === 'reset' &&
    css`
      color: ${({ theme }) => theme.color.resetButtonFont};
    `}
`;
