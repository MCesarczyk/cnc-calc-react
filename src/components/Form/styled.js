import styled from "styled-components";

export const FormWrapper = styled.form`
    color: ${({ theme }) => theme.color.primary};
    margin-bottom: 2rem;
`;

export const Fieldset = styled.fieldset`
    border-radius: 2rem;
    padding: 1rem;
`;

export const Legend = styled.legend`
    text-align: center;
    margin: 0.5rem
`;

export const FieldsContainer = styled.div`
    display: grid;
    justify-content: center;
    gap: 1rem;
    margin-bottom: 1.5rem;

    @media (max-width: 480px) {
        gap: 0.5rem;
    }
`;