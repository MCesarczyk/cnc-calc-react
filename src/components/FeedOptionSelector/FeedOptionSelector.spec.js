import React from "react";
import renderer from "react-test-renderer";
import { ThemeProvider } from "styled-components";
import { theme } from "../../theme";
import "jest-styled-components";
import RadioButton from "../RadioButton";
import { RadioButtonContainer } from "./styled";
import { Input } from "../Input";
import { FieldWrapper } from "../FieldWrapper";

test("FeedOptionSelector component rendering properly", () => {
  const component = renderer.create(
    <ThemeProvider theme={theme} >
      <FieldWrapper>
        <RadioButtonContainer>
          <RadioButton />
        </RadioButtonContainer>
        <Input />
      </FieldWrapper>
    </ThemeProvider>
  );

  let tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});