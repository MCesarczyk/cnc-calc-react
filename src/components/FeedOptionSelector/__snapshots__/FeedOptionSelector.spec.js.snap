// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedOptionSelector component rendering properly 1`] = `
.c2 {
  vertical-align: sub;
  font-size: 0.7em;
  margin-right: 0.15em;
}

.c1 {
  display: flex;
  gap: 10px;
}

.c0 {
  display: flex;
  gap: 0.5rem;
  align-items: center;
  justify-content: space-between;
}

.c3 {
  min-width: 16rem;
  border: solid 1px #1e3246;
  border-radius: 5px;
  padding: 5px;
}

.c3:invalid {
  background-color: beige;
}

.c3:disabled {
  background-color: #ddd;
}

@media (max-width: 480px) {
  .c1 {
    align-self: flex-start;
  }
}

@media (max-width: 480px) {
  .c0 {
    flex-direction: column;
    justify-content: center;
    gap: 0.25rem;
  }
}

@media (max-width: 480px) {
  .c3 {
    width: 100%;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <input
      checked={true}
      data-testid="undefined-radio"
      type="radio"
    />
    <label>
      <span
        className="c2"
      />
    </label>
  </div>
  <input
    className="c3"
  />
</div>
`;
